src/arch/xen/event.c

```diff
@@ -126,6 +127,9 @@ void do_hypervisor_callback(struct pt_regs *regs)
 
 			/* Combine the two offsets to get the port */
 			evtchn_port_t port = (pending_selector << 5) + event_offset;
+
+			irq_dispatch(port);
```

Comment:
My bad, I suggested to call just `irq_dispatch`. Actually, interrupt handling scheme is rather good defined, for example, src/arch/x86/kernel/irq_handler.c. This required to be extended to that in order to support nested interrupts